core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -102,11 +111,27 @@ public Optional<Result> get(Get originalGet) throws CrudException {
       key = new Snapshot.Key(get);
     }
 
-    readUnread(key, get);
-
-    return snapshot
-        .getResult(key, get)
-        .map(r -> new FilteredResult(r, originalProjections, metadata, isIncludeMetadataEnabled));
+    if (isSnapshotReadRequired()) {
+      readUnread(key, get);
+      return snapshot
+          .getResult(key, get)
+          .map(r -> new FilteredResult(r, originalProjections, metadata, isIncludeMetadataEnabled));
+    } else {
+      Optional<TransactionResult> result = read(key, get);
+      if (get.getConjunctions().isEmpty()) {
+        return snapshot
+            .mergeResult(key, result)
+            .map(
+                r ->
+                    new FilteredResult(r, originalProjections, metadata, isIncludeMetadataEnabled));
+      } else {
```

Comment:
How about handling null/empty conjunction in the `mergeResult` method to unify the operations in the if/else clauses?

Reply:
Fixed in cefe83b4c8fb7d9d8ec3ca29740ec396b107d770. Thanks!